Add controller for MksTaskRun

avinal opened the pull request:
# Add controller for MksTaskRun

- Listens for MksTaskRun resource being created, updated or deleted
- Adds corresponding TaskRun using Tekton API

# How to test

- Build and run the executable using steps in README.md
- Apply the CRDs and example in the config folder.
- Add  a new MksTaskRun resource and see for logs

# Related issues

Fixes #13

avinal commented:
Hello @piyush-garg, please review.

ramessesii2 commented on pkg/controllers/mkstaskrun/mkstaskrun.go:
```diff
@@ -0,0 +1,68 @@
+	// 	TaskRef:     &v1beta1.TaskRef{Name: mtr.Spec.TaskRef.Name},
+	// 	PodTemplate: &v1beta1.PodTemplate{},
+	// }
+	return res
```
a `return` is redundant

piyush-garg commented on pkg/controllers/mkstaskrun/mkstaskrun.go:
```diff
@@ -0,0 +1,68 @@
+
+	return res
+
+	// res := &v1beta1.TaskRun{}
```
remove these comments and return written twice

piyush-garg commented on pkg/controllers/mkstaskrun/controller.go:
```diff
@@ -0,0 +1,69 @@
+func addController(obj interface{}) {
+	fmt.Println("MksTaskRun has been created")
+
+	tp := &tconfig.TektonParam{}
```
why you need these ?

avinal replied:
Tekton configuration wll be needed to communicate with Tekton API. I guess this is redundant and the configuration should be passed to NewController func instead of having one for each controller.

avinal committed:
Add controller for MksTaskRun

    - Adds controller code that can listen for new resources on mks-server
    and create corresponding Tekton resource using Tekton API
    - Add a basic driver code
    - Fixes #13

Signed-off-by: Avinal Kumar <avinkuma@redhat.com>

avinal committed:
Add example config files

    - Add MksTaskRun definition and exmaple tekton Task
    - Update go dependencies

Signed-off-by: Avinal Kumar <avinkuma@redhat.com>
